pkg/cluster/internal/providers/provider/common/proxy.go

```diff
@@ -50,6 +53,18 @@ func getProxyEnvs(cfg *config.Cluster, getEnv func(string) string) map[string]st
 			envs[strings.ToLower(name)] = val
 		}
 	}
+
+	// if we didn't detect proxy environment variable settings, probe docker
+	// we prefer the proxy env to allow more explicit per-invocation override
+	// of proxy settings
+	if len(envs) < 1 {
+		e, err := getProxyEnvFromDocker(cmder)
```

Comment:
TODO: probably re-work this so some providers don't do this? need to think about this more, but that shouldn't be hard and it's academic at the moment.

Reply:
can it be an interface `getProxyEnvFromProvider`?